src/src_main/crypto_tc.c

```diff
@@ -891,13 +890,12 @@ int32_t Crypto_TC_ProcessSecurity(uint8_t* ingest, int *len_ingest, TC_t* tc_sdl
     {
         // TODO - implement non-AEAD algorithm logic
 
-        if(sa_service_type == SA_ENCRYPTION || sa_service_type == SA_AUTHENTICATED_ENCRYPTION)
-        { 
+        if(sa_service_type == SA_ENCRYPTION) 
```

Comment:
The case is still valid for Encryption of Authenticated Encryption (even though we don't support any non-AEAD encryption algorithms yet). Removing the or condition is wrong.

Reply:
Hmm, AEAD has its own aead_decrypt ca. line 869 and that function also checks validation as part of AEAD decrypt

Reply:
I think I'm tracking better now - disregard the previous comment. You're talking a specific case as mentioned in the next comment

Reply:
Correct. You're confusing service type with encryption/authentication algorithm workings... You can still have Authenticated Encryption even with non-AEAD algorithms (which is what this is doing here)